File: hcsr04.c
#include <fcntl.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <linux/ioctl.h>

struct Data{

  long long int value;
};
int init()
{

  int fd = open("/dev/hcsr04", O_RDWR);
  return fd;
}

float measure(int fd){
  
  struct Data data;
  ioctl(fd, _IO(0, 3), &data);
  
  return (float)((data.value)/58);
}

void finalization(int fd){
  close(fd);
}

File: hcsr04.py
import fnctl
import os

class Data:
    value: int = None

def init():
    device_file = '/dev/hcsr04'
    fd = os.open(device_file, os.O_RDWR)
    return fd

def measure(fd):
    data = Data()
    fnctl.ioctl(fd, _IO(0, 3), data)

    return (float)((data.value)/58)

def finalization(fd):
    os.close(fd)